components/dashboard/src/workspaces/WorkspaceEntry.tsx

```diff
@@ -46,14 +46,21 @@ export function WorkspaceEntry({desc, model}: {desc: WorkspaceInfo, model: Works
         pathname: '/start/',
         hash: '#' + ws.id
     });
-    const downloadURL = new GitpodHostUrl(window.location.href).with({ 
-        pathname: `/workspace-download/get/${ws.id}` 
+    const downloadURL = new GitpodHostUrl(window.location.href).with({
+        pathname: `/workspace-download/get/${ws.id}`
     }).toString();
     const menuEntries: ContextMenuEntry[] = [
         {
             title: 'Open',
             href: startUrl.toString()
-        },
+        }];
+    if (state === 'running') {
+        menuEntries.push({
+            title: 'Stop',
+            onClick: () => getGitpodService().server.stopWorkspace(ws.id)
+        });
+    }
+    menuEntries.push(
         {
             title: 'Download',
             href: downloadURL
```

Comment:
**issue**: Download is not working yet, right? Is this related to https://github.com/gitpod-io/gitpod/issues/3275?